common_types/src/bitset.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022-2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! Simple BitSet implementation.
+
+/// A basic implementation supporting read/write.
+#[derive(Debug, Default, Clone)]
+pub struct BitSet {
+    /// The bits are stored as bytes in the least significant bit order.
+    buffer: Vec<u8>,
+    /// The number of real bits in the `buffer`
+    num_bits: usize,
+}
+
+impl BitSet {
+    /// Initialize a unset [`BitSet`].
+    pub fn new(num_bits: usize) -> Self {
```

Comment:
I think may be we can provide an another constructor with all bits set to 1, since in some case, there is hardly no null values at all.